packages/core/src/main/conversation/ConversationService.ts

```diff
@@ -144,6 +146,28 @@ class ConversationService {
     };
   }
 
+  private async sendConfirmationEphemeral(
+    conversationId: string,
+    payloadBundle: PayloadBundleOutgoingUnsent
+  ): Promise<PayloadBundleOutgoing> {
+    const confirmationContent = payloadBundle.content as ConfirmationContent;
+
+    const confirmationObject: PayloadBundleOutgoingUnsent = {
+      ...payloadBundle,
+      type: GenericMessageType.CONFIRMATION,
+    };
+
+    await this.send(conversationId, confirmationObject);
+    await this.deleteMessageEveryone(conversationId, confirmationContent.confirmMessageId);
```

Comment:
Humm there seems to be a flaw in the process here. 
Basically what this code means is that the first that receives the ephemeral message will actually try to delete it for everyone.

In the webapp, what we do is sending this deletion message (in the same conversation) but only targeted (/encrypted) for the person who actually sent the message. 

(see https://github.com/wireapp/wire-webapp/blob/e1357df97249e328cb24e6977444af7417f0513c/app/script/conversation/ConversationRepository.js#L2551)

As you can see, in the webapp, we have a parameter for the `delete_message_everyone` where we can specify a bunch of user ids to send the message to.

Maybe it's the flow to use in that case, maybe we should invent something new, but we should not send the `delete` message to everyone in the convo. 

ping @gregor @bennyn

Reply:
That's technically correct, but we never implemented encrypting messages for certain people. So for now we have to send the `MessageDelete` to everyone in the group. That's a bit of overhead but every client needs to be able to handle this anyway.